lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
+      s.register
+      s
+    end
+    it "returns from run" do
+      consumer_thread = Thread.new(queue) { |queue| loop { queue.pop } }
+      plugin_thread = Thread.new(subject, queue) { |subject, queue| subject.run(queue) }
+      # the run method is a long lived one, so it should still be running after "a bit"
+      sleep 0.5
```

Comment:
absolutely makes sense, the problem is implementing it..
Here's one example: consider a possible rabbitmq input plugin that does:

```ruby
def run(queue)
  rabbitmq_server.new(@options).subscribe do |event|
    queue << LogStash::Event.new(event)
  end
end
```

There's no place to put the `ready?` anywhere once the code goes into the client library.
And actually, `ready?` is only really useful for testing, in a real world situation we don't care that much: "eventually" the plugin will be ready and start producing events, but we don't need to know exactly when.